source/dub/generators/build.d

```diff
@@ -48,7 +48,8 @@ string computeBuildName(string config, in GeneratorSettings settings, const stri
 	addHash(settings.platform.compilerVersion);
 	const hashstr = Base64URL.encode(hash.finish()[0 .. $ / 2]).stripRight("=");
 
-	return format("%s-%s-%s", config, settings.buildType, hashstr);
+	
+	return format("%s-%s-%s-%s", config, settings.buildType, settings.recipeName, hashstr);
```

Comment:
we often had issues with length limits here on windows, can you in the default case (dub.sdl/dub.json) make it so the cache name is changed, only changing when it's actually modified? This would also keep existing caches valid.

Reply:
It verifies now if there is some recipe name before returning